modules/network/main.tf

```diff
@@ -0,0 +1,177 @@
+resource "aws_vpc" "main" {
```

Comment:
Instead of building all VPC components manually like this we can reuse the `VPC module` from AWS :D 

[terraform-aws-modules/vpc/aws](https://registry.terraform.io/modules/terraform-aws-modules/vpc/aws/latest)

It's faster, safer than our build everything, think about it's like a `library` in the language

Reply:
Yeah wasn't aware of that module. I know we can use public modules in the `source` but didn't think to re-use that.

A benefit is it allowed me to learn more of the components I need and how to use them in my own infrastructure. If I used this I would still have to understand the inner workings and how to use it to begin with